tests/test_project_management.py

```diff
@@ -523,14 +523,33 @@ def _assert_request_is_correct(
         assert request.url == expected_url
         client_version = 'Python/Admin/{0}'.format(firebase_admin.__version__)
         assert request.headers['X-Client-Version'] == client_version
-        assert request._extra_kwargs['timeout'] == pytest.approx(
-            _http_client.DEFAULT_TIMEOUT_SECONDS, 0.001)
         if expected_body is None:
             assert request.body is None
         else:
             assert json.loads(request.body.decode()) == expected_body
 
 
+class TestTimeout(BaseProjectManagementTest):
+
+    def test_default_timeout(self):
+        app = firebase_admin.get_app()
+        project_management_service = project_management._get_project_management_service(app)
+        assert project_management_service._client.timeout == _http_client.DEFAULT_TIMEOUT_SECONDS
+
+    @pytest.mark.parametrize('timeout', [4, None])
+    def test_custom_timeout(self, timeout):
+        options = {
+            'httpTimeout': timeout,
+            'projectId': 'test-project-id'
+        }
+        app = firebase_admin.initialize_app(testutils.MockCredential(), options, 'timeout-app')
+        try:
+            project_management_service = project_management._get_project_management_service(app)
+            assert project_management_service._client.timeout == timeout
+        finally:
+            firebase_admin.delete_app(app)
```

Comment:
BaseProjectManagementTest seems to call testutils.cleanup_apps() in it's teardown() method. If you're willing to rely on that, you could drop the try, finally, and delete_app() lines. Presumably you'd nave to not hard-code the app name ("timeout-app") but instead use something based on the test parameter or something like that instead. Optional.

I have no objection to leaving it in either, but it's odd to see the app cleaned up in this test case, but not the one above.

Reply:
Done